### PR TITLE
[MIG] project_task_materials: Migrated to 10.0

### PR DESCRIPTION
**Pull request by rigo1985**

### CODE REVIEW

File: project_task_materials/models/project.py [hunk 1]
```diff
@@ -1,40 +1,31 @@
...
 
 class Task(models.Model):
     _inherit = "project.task"
-    material_ids = fields.One2many(
-        comodel_name='project.task.materials', inverse_name='task_id',
-        string='Materials used')
+
+    material_ids = fields.One2many(comodel_name='project.task.materials',
+                                   inverse_name='task_id',
+                                   string='Materials used')
```
Comment (dreispt) [inline project_task_materials/models/project.py:15]: Please avoid code style only changes.
Both ways are PEP8 compliant, but when so IMO the original author's preference should be respected.

File: project_task_materials/models/project.py [hunk 2]
```diff
@@ -1,40 +1,31 @@
...
-    task_id = fields.Many2one(
-        comodel_name='project.task', string='Task', ondelete='cascade',
-        required=True)
-    product_id = fields.Many2one(
-        comodel_name='product.product', string='Product', required=True)
+
+    task_id = fields.Many2one(comodel_name='project.task', required=True,
+                              string='Task', ondelete='cascade')
+    product_id = fields.Many2one(comodel_name='product.product',
+                                 string='Product', required=True)
```
Comment (dreispt) [inline project_task_materials/models/project.py:25]: Same as above-

File: project_task_materials/views/project_view.xml [hunk 3]
```diff
@@ -1,25 +1,24 @@
...
-              <field name="product_id"/>
-              <field name="quantity"/>
-            </tree>
-          </field>
+        <field name="name">project.task.form.materials</field>
+        <field name="model">project.task</field>
+        <field name="inherit_id" ref="project.view_task_form2"/>
+        <field name="arch" type="xml">
+            <notebook position="inside">
+                <page string="Materials used" attrs="{'invisible': [('project_id', '=', False)]}">
```
Comment (dreispt) [inline project_task_materials/views/project_view.xml:10]: When completing the task, the user will record all time and materials spent.
IMO it is a better UI to have the Materials spend list in the same place as the work lines.

File: project_task_materials/models/project.py [hunk 4]
```diff
@@ -1,40 +1,31 @@
...
     quantity = fields.Float(string='Quantity')
+    uom_id = fields.Many2one('product.uom', string='Unit of Measure')
+
```
Comment (dreispt) [inline project_task_materials/models/project.py:28]: Good addition, but shouldn't it have a default value?

File: project_task_materials/views/project_view.xml (hunk 3 above)
Comment (rigo1985) [reply project_task_materials/views/project_view.xml:10]: @dreispt  I think it's easier for the end user. I consider that later the possibility of importing the materials to be used that can be many should be added, so I add a new TAB. 
cc @Tecnativa

File: project_task_materials/models/project.py (hunk 4 above)
Comment (rigo1985) [reply project_task_materials/models/project.py:28]: @dreispt This field obtains the value of the unit of measure of the product.

Comment (dreispt) [reply project_task_materials/models/project.py:28]: OK

File: project_task_materials/views/project_view.xml (hunk 3 above)
Comment (dreispt) [reply project_task_materials/views/project_view.xml:10]: This is not my opinion, but I'm open to your view if I see other supporting it.

Comment (dreispt) [reply project_task_materials/views/project_view.xml:10]: I tried this PR on the OCA Runbot, and the material list was not editable.
I'm not sure if its a problem with Runbot or with the module - can you check?

**Comment by pedrobaeza** [comment]
Superseeded by #251

**Pull request closed without merging**